ql/src/java/org/apache/hadoop/hive/ql/parse/repl/metric/MetricSink.java

```diff
@@ -116,14 +116,17 @@ public void run() {
           int totalMetricsSize = metrics.size();
           List<ReplicationMetrics> replicationMetricsList = new ArrayList<>(totalMetricsSize);
           ObjectMapper mapper = new ObjectMapper();
+          MessageEncoder encoder = MessageFactory.getDefaultInstanceForReplMetrics(conf);
           for (int index = 0; index < totalMetricsSize; index++) {
             ReplicationMetric metric = metrics.removeFirst();
             ReplicationMetrics persistentMetric = new ReplicationMetrics();
             persistentMetric.setDumpExecutionId(metric.getDumpExecutionId());
             persistentMetric.setScheduledExecutionId(metric.getScheduledExecutionId());
             persistentMetric.setPolicy(metric.getPolicy());
-            persistentMetric.setProgress(mapper.writeValueAsString(metric.getProgress()));
+            persistentMetric.setProgress(
+                    encoder.getSerializer().serialize(mapper.writeValueAsString(metric.getProgress())));
```

Comment:
We will end up doing serialization twice. Can you pls move the column limit check right here so that we do it just once?